integration-tests/tests/test_download_file.rs

```diff
@@ -0,0 +1,101 @@
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::{Read, Write};
+use std::path::Path;
+use testcontainers::*;
+
+#[derive(Default, Debug, Clone)]
+pub struct NginxArgs {}
+
+impl IntoIterator for NginxArgs {
+    type Item = String;
+    type IntoIter = ::std::vec::IntoIter<String>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        vec![].into_iter()
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct Nginx<'a> {
+    arguments: NginxArgs,
+    env_vars: HashMap<String, String>,
+    version: String,
+    static_content_path: Option<&'a Path>,
+}
+
+impl Image for Nginx<'_> {
+    type Args = NginxArgs;
+    type EnvVars = HashMap<String, String>;
+    type Volumes = HashMap<String, String>;
+    type EntryPoint = std::convert::Infallible;
+
+    fn descriptor(&self) -> String {
+        format!("nginx:{}", self.version)
+    }
+
+    fn wait_until_ready<D: Docker>(&self, _container: &Container<'_, D, Self>) {}
+
+    fn args(&self) -> Self::Args {
+        self.arguments.clone()
+    }
+
+    fn env_vars(&self) -> Self::EnvVars {
+        self.env_vars.clone()
+    }
+
+    fn volumes(&self) -> Self::Volumes {
+        let mut volumes = HashMap::new();
+        match self.static_content_path {
+            None => {}
+            Some(static_content_path) => {
+                volumes.insert(
+                    static_content_path.to_str().unwrap().into(),
+                    "/usr/share/nginx/html".into(),
+                );
+                ()
+            }
+        }
+        volumes
+    }
+
+    fn with_args(self, arguments: Self::Args) -> Self {
+        Self { arguments, ..self }
+    }
+}
+
+#[test]
+fn test_file_download() {
+    // Set up a temporary folder holding static file that is going to be served by nginx
+    let tmp_static_content_dir = tempfile::tempdir().unwrap();
+    let static_file_name: String = "file_to_download".into();
+    let tmp_static_file_path = tmp_static_content_dir.path().join(&static_file_name);
+    // Create a static file that holds 1MB of zeros
+    let mut tmp_static_file = File::create(tmp_static_file_path).unwrap();
+    let zeros: [u8; 1024 * 1024] = [0; 1024 * 1024];
+    tmp_static_file.write_all(&zeros).unwrap();
+
+    let mut nginx = Nginx::default();
+    nginx.version = "1.21".to_string();
+    nginx.static_content_path = Some(tmp_static_content_dir.path());
+
+    let docker = clients::Cli::default();
+    let container = docker.run(nginx);
```

Comment:
Well, this adds a new host system dependency on having Docker and being able to run it (e.g. it is problematic if you run tests inside a Docker container)

Reply:
Hmm, I didn't think using testcontainer-rs would be introducing a new host system dependency since Docker seems to be actively used in Python tests (like [here](https://github.com/near/nearcore/blob/master/pytest/tests/stress/hundred_nodes/start_100_nodes.py)). That being said, setting up a hyper server that just serves a byte array is actually simpler code-wise, so I went with that instead. PTAL.